src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/Infrastructure/AspNetCoreVirtualCharSequenceExtensions.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.EmbeddedLanguages;
+
+namespace Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
+
+internal static class AspNetCoreVirtualCharSequenceExtensions
+{
+    public static AspNetCoreVirtualChar First(this AspNetCoreVirtualCharSequence virtualChars)
+    {
+        return virtualChars[0];
+    }
+
+    public static AspNetCoreVirtualChar Last(this AspNetCoreVirtualCharSequence virtualChars)
+    {
+        return virtualChars[virtualChars.Length - 1];
```

Comment:
```suggestion
        return virtualChars[^1];
```

Reply:
Can't do this because it's netstandard2.0 and don't have internal code for Range type in the project.